src/Virtuozzo/Helper/XMLCommand.php

```diff
@@ -0,0 +1,578 @@
+<?php
+
+namespace Upmind\ProvisionProviders\Servers\Virtuozzo\Helper;
+
+use DOMDocument;
+use DOMElement;
+use Exception;
+use Upmind\ProvisionBase\Exception\ProvisionFunctionError;
+
+class XMLCommand
+{
+    public const XML_VERSION = '1.0';
+
+    public const XML_ENCODING = 'UTF-8';
+    protected string $apiVersion;
+
+    protected DOMDocument $domDocument;
+
+    protected string $interface;
+
+    protected array $realms = [
+        'system' => '00000000-0000-0000-0000-000000000000',
+    ];
+
+    protected array $rootElementAttrs = [
+        'xmlns:xsi' => 'http://www.w3.org/2001/XMLSchema-instance',
+        'xmlns:ns2' => 'http://www.swsoft.com/webservices/vzl/4.0.0/types',
+        'xmlns:ns3' => 'http://www.swsoft.com/webservices/vzp/4.0.0/vzptypes',
+        'xmlns:ns4' => 'http://www.swsoft.com/webservices/vza/4.0.0/vzatypes'
+    ];
+
+    public function __construct(
+        string $version = '7.0.0',
+        string $encoding = self::XML_ENCODING,
+        string $interface = 'vzpenvm'
+    )
+    {
+        $this->domDocument = new DOMDocument(self::XML_VERSION, $encoding);
+
+        $this->apiVersion = $version;
+
+        $this->setInterface($interface);
+    }
+
+    public function setInterface(string $type)
+    {
+        $this->interface = $type;
+    }
+
+    /**
+     * @param DOMElement $body
+     * @param bool $target
+     * @return string
+     * @throws ProvisionFunctionError
+     */
+    public function build(DOMElement $body, bool $target = true): string
+    {
+        try {
+            $rootWrapper = $this->makeRootWrapper();
+            $this->getDomDocument()->appendChild($rootWrapper);
+
+            if ($target) {
+                $rootWrapper->appendChild($this->createElement('target', $this->interface));
+            }
+
+            $rootWrapper->appendChild($body);
+
+            return $this->toString();
+        } catch (Exception $e) {
+            throw ProvisionFunctionError::create($e->getMessage());
```

Comment:
Please always remember to chain the original exception in the re-thrown exception, so that logs contain the trace of the original underlying error